### PR TITLE
chore: update storybook

### PR DESCRIPTION
**Pull request by gabyzif**
## Summary
This PR updates Storybook and it's dependencies

## Testing Plan

- [x] Was this tested locally? If not, explain why.
- {explain how this has been tested, and what, if any, additional testing should be done}

 
- Closes https://go.mparticle.com/work/UNI-1021

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -30,10 +30,10 @@ const config: StorybookConfig & StorybookVitePlugins = {
         },
       },
     },
+    '@chromatic-com/storybook',
```
Comment (tibuurcio) [inline .storybook/main.ts:33]: Why do we need chromatic? I was thinking we would get rid of it now that we have PR envs?

Comment (gabyzif) [reply .storybook/main.ts:33]: I was thinking the same thing

File: .storybook/main.ts [hunk 2]
```diff
@@ -49,6 +48,10 @@ const config: StorybookConfig & StorybookVitePlugins = {
   core: {
     disableTelemetry: true, // requested by security
   },
+
+  typescript: {
+    reactDocgen: 'react-docgen-typescript',
+  },
```
Comment (tibuurcio) [inline .storybook/main.ts:54]: Why is that needed?

File: .storybook/preview.tsx [hunk 3]
```diff
@@ -52,6 +52,8 @@ const preview: Preview = {
       },
     },
   },
+
+  tags: ['autodocs'],
```
Comment (tibuurcio) [inline .storybook/preview.tsx:56]: same as the above

Comment (gabyzif) [reply .storybook/preview.tsx:56]: This happened when upgrading SB

Before we we had the autodocs placed on main here:

Screenshot 2024-10-24 at 5 37 53 PM

when I was upgrading SB this promt:

Screenshot 2024-10-24 at 5 24 24 PM

so basically it did the migration to the new version automatically.


more info here: [migration docs](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#autodocs-changes )

File: .storybook/main.ts (hunk 2 above)
Comment (gabyzif) [reply .storybook/main.ts:54]: When doing the migration this was prompt:

Screenshot 2024-10-24 at 5 45 54 PM

So basically SB swiched to react-docgen but we had react-docgen-typescript on place and by doing this we are keeping it and not moving to react-docgen.

[More info here](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-docgen-component-analysis-by-default 
)
